pikitlib.py

```diff
@@ -16,70 +34,203 @@ def __init__(self, id):
         # get and save button state
         
         # A-0,B-1,X-2,Y-3
-        self.lastButtonValues = []
-        self.lastButtonValues.append(self.nt.getBoolean("Button0", False))
-        self.lastButtonValues.append(self.nt.getBoolean("Button1", False))
-        self.lastButtonValues.append(self.nt.getBoolean("Button2", False))
-        self.lastButtonValues.append(self.nt.getBoolean("Button3", False))
 
+        # A, B, X, Y, L bumpter, R bumber, back, start, big shinny button
+        self.buttons = [0,0,0,0,0,0,0,0]
+
+        # LHand Y, LHand X, L trigger, RHand X, RHand Y, R trigger
+        self.axis_values = [0,0,0,0,0,0]
 
-    def getButton(self, v) -> bool:
-        newB = self.nt.getBoolean("Button" + str(v), False)
-        self.lastButtonValues[v] = newB
+
+        for i in range(len(self.buttons)):
```

Comment:
I think we can be clever and only read the buttons and axis only once every cycle.    We should talk about this.      I'm not certain how to pull that off.     Or maybe we check to see if anything is changed in the subtree and then load both the arrays?     Can we have a changehandler that can do that?         So, we assume that the state is up to date because we know the changeHandler would have gotten called and updated everything.     Can you think of any reason not to do it that way?       I haven't dug deep into the wpilib code to find out how the state makes it.   I'm assuming it is some part of the driverstation protocol, which I have no interest in reverse engineering...   Talk to me.